momepy/functional/_intensity.py

```diff
@@ -0,0 +1,50 @@
+import geopandas as gpd
+from geopandas import GeoDataFrame, GeoSeries
+from libpysal.graph import Graph
+from packaging.version import Version
+from pandas import Series
+from shapely import buffer, get_num_interior_rings, unary_union
+
+__all__ = ["courtyards"]
+
+GPD_013 = Version(gpd.__version__) >= Version("0.13")
+
+
+def courtyards(geometry: GeoDataFrame | GeoSeries, graph: Graph) -> Series:
+    """Calculate the number of courtyards within the joined structure.
+
+    Adapted from :cite:`schirmer2015`.
+
+    Parameters
+    ----------
+    geometry : GeoDataFrame
+        A GeoDataFrame containing objects to analyse.
+    graph : libpysal.graph.Graph
+        A spatial weights matrix for the geodataframe,
+        it is used to denote adjacent buildings.
+    Returns
+    ----------
+    series : Series
+        A Series containing the resulting values.
+
+    Examples
+    --------
+    >>> courtyards = mm.calculate_courtyards(buildings_df, graph)"""
```

Comment:
```suggestion
    >>> courtyards = mm.calculate_courtyards(buildings_df, graph)
    """
```

We will eventually want this executable but that can be done together with other examples.